internal/controller/clientpool/clientpool.go

```diff
@@ -44,11 +53,30 @@ func (cp *ClientPool) GetSDKClient(key ClientPoolKey) (sdkclient.Client, bool) {
 	cp.mux.RLock()
 	defer cp.mux.RUnlock()
 
-	c, ok := cp.clients[key]
-	if ok {
-		return c, true
+	info, ok := cp.clients[key]
+	if !ok {
+		return nil, false
 	}
-	return nil, false
+
+	// Check if any certificate is expired
+	if info.TLS != nil {
+		for _, cert := range info.TLS.Certificates {
+			if len(cert.Certificate) == 0 {
+				continue
+			}
+			expired, err := cp.isCertificateExpiredSoon(cert.Certificate[0], 30*time.Minute)
+			if err != nil {
+				cp.logger.Error("Error checking certificate expiration", "error", err)
+				return nil, false
+			}
+			if expired {
+				cp.logger.Info("Certificate expired", "expiry", cert.Leaf.NotAfter)
```

Comment:
```suggestion
			almostExpired, err := cp.isCertificateExpiredSoon(cert.Certificate[0], 30*time.Minute)
			if err != nil {
				cp.logger.Error("Error checking certificate expiration", "error", err)
				return nil, false
			}
			if almostExpired {
				cp.logger.Info("Certificate expiring soon", "expiry", cert.Leaf.NotAfter)
```

Reply:
wait actually this isn't quite accurate, cuz the cert could also be expired. let me edit